shell/platform/fuchsia/flutter/component_v1.cc

```diff
@@ -62,15 +65,27 @@ std::string DebugLabelForUrl(const std::string& url) {
 void ComponentV1::ParseProgramMetadata(
     const fidl::VectorPtr<fuchsia::sys::ProgramMetadata>& program_metadata,
     std::string* data_path,
-    std::string* assets_path) {
+    std::string* assets_path,
+    int64_t* old_gen_heap_size) {
   if (!program_metadata.has_value()) {
     return;
   }
+
+  *old_gen_heap_size = kNoOldGenHeapSizeSpecified;
+
   for (const auto& pg : *program_metadata) {
     if (pg.key.compare(kDataKey) == 0) {
       *data_path = "pkg/" + pg.value;
     } else if (pg.key.compare(kAssetsKey) == 0) {
       *assets_path = "pkg/" + pg.value;
+    } else if (pg.key.compare(kOldGenHeapSizeKey) == 0) {
```

Comment:
Use "args" instead of a custom key here, otherwise this will complicate converting to the ELF runner later:  https://fuchsia.dev/reference/cml?hl=en#elf-runners

The arg can look like --old_gen_heap_size=<size>